java-symbolic-execution/src/main/java/org/sonar/java/se/checks/XmlValidatedSighatureCheck.java

```diff
@@ -0,0 +1,221 @@
+
+  @Override
+  public ProgramState checkPostStatement(CheckerContext context, Tree syntaxNode) {
+    PostStatementVisitor visitor = new PostStatementVisitor(context);
+    syntaxNode.accept(visitor);
+    return visitor.programState;
+  }
+
+  private static class PostStatementVisitor extends CheckerTreeNodeVisitor {
+
+    private PostStatementVisitor(CheckerContext context) {
+      super(context.getState());
+    }
+
+    @Override
+    public void visitNewClass(NewClassTree tree) {
+      if (DOM_VALIDATE_CONTEXT_CONSTRUCTOR.matches(tree)) {
+        programState = programState.addConstraint(programState.peekValue(0), DomSecureValidation.DISABLED);
+      }
+    }
+
+    @Override
+    public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+      ProgramState.SymbolicValueSymbol peek = programState.peekValueSymbol();
+      Symbol symbol = peek.symbol();
+      SymbolicValue sv = peek.symbolicValue();
+      if (symbol != null && sv instanceof DomValidateContextSymbolicValue) {
+        ((DomValidateContextSymbolicValue) sv).setField(ProgramState.isField(symbol));
+      }
+    }
+  }
+
+  @Override
+  public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
+    PreStatementVisitor visitor = new PreStatementVisitor(context);
+    syntaxNode.accept(visitor);
+    return visitor.programState;
+  }
+
+  private class PreStatementVisitor extends CheckerTreeNodeVisitor {
+
+    private static final String SECURE_VALIDATION_PROPERTY = "org.jcp.xml.dsig.secureValidation";
+    private final CheckerContext context;
+
+    private PreStatementVisitor(CheckerContext context) {
+      super(context.getState());
+      this.context = context;
+    }
+
+    @Override
+    public void visitNewClass(NewClassTree newClass) {
+      if (DOM_VALIDATE_CONTEXT_CONSTRUCTOR.matches(newClass)) {
+        context.getConstraintManager().setValueFactory(() -> new DomValidateContextSymbolicValue(newClass.identifier()));
+      }
+    }
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree mit) {
+      if (SET_PROPERTY.matches(mit)) {
+        Arguments args = mit.arguments();
+        if (!SECURE_VALIDATION_PROPERTY.equalsIgnoreCase(args.get(0).asConstant(String.class).orElse(""))) {
+          // if we do not resolve correctly the property name, this can lead to FPs as we would not discard the constraint
+          return;
+        }
+        SymbolicValue domSv = programState.peekValue(args.size());
+        if (programState.getConstraint(programState.peekValue(0), BooleanConstraint.class) == BooleanConstraint.FALSE) {
+          programState = programState.addConstraint(domSv, DomSecureValidation.EXPLICITLY_DISABLED);
+        } else {
+          // directly set to true, or no idea what the value could be, better say nothing
+          programState = programState.removeConstraintsOnDomain(domSv, DomSecureValidation.class);
+        }
+      } else if (XML_SIGNATURE_VALIDATE.matches(mit)) {
```

Comment:
I checked a bit the usages of `XMLValidateContext`, and it seems that there are other ways to use the context. For example:

- [XMLSignature](https://docs.oracle.com/javase/7/docs/api/javax/xml/crypto/dsig/XMLSignature.html#validate(javax.xml.crypto.dsig.XMLValidateContext))

```
XMLSignature signature;
signature.getSignatureValue().validate(context);
```

- [Reference](https://docs.oracle.com/javase/7/docs/api/javax/xml/crypto/dsig/Reference.html#validate(javax.xml.crypto.dsig.XMLValidateContext))
```
org.apache.jcp.xml.dsig.internal.dom.DOMReference dom;
dom.validate(context);
```

Maybe we should double-check with Appsec if we should report this kind of calls.